### PR TITLE
go: detect structs/interfaces & parse struct members and function arguments

### PR DESCRIPTION
**Pull request by techee**
More info in the individual commits.

### CODE REVIEW

**Comment by masatake** [comment]
Valgrind is available on your PC?

$ make check UNITS=go-torture  VG=1; cat Units/go-torture.d/VALGRIND.tmp
...
==13417== 192 (80 direct, 112 indirect) bytes in 2 blocks are definitely lost in loss record 3 of 3
==13417==    at 0x4A06BCF: malloc (in /usr/lib64/valgrind/vgpreload_memcheck-amd64-linux.so)
==13417==    by 0x42C035: eMalloc (routines.c:216)
==13417==    by 0x416462: copyToken (go.c:164)
==13417==    by 0x416787: parseStructMembers (go.c:650)
==13417==    by 0x416787: parseConstTypeVar (go.c:747)
==13417==    by 0x416984: parseGoFile (go.c:792)
==13417==    by 0x416984: findGoTags (go.c:813)
==13417==    by 0x424B1B: createTagsForFile (parse.c:1570)
==13417==    by 0x42615E: createTagsWithFallback (parse.c:1614)
==13417==    by 0x42615E: parseFile (parse.c:1673)
==13417==    by 0x41C634: createTagsForEntry (main.c:241)
==13417==    by 0x402B6F: createTagsForArgs (main.c:286)
==13417==    by 0x402B6F: makeTags (main.c:432)
==13417==    by 0x402B6F: main (main.c:517)
...
==13417== LEAK SUMMARY:
==13417==    definitely lost: 80 bytes in 2 blocks
==13417==    indirectly lost: 112 bytes in 4 blocks
==13417==      possibly lost: 0 bytes in 0 blocks
==13417==    still reachable: 0 bytes in 0 blocks
==13417==         suppressed: 0 bytes in 0 blocks

**Comment by techee** [comment]
Oops, thanks, will fix.

**Comment by techee** [comment]
The memory leak fix can be squashed with the previous commits, posting separately to make it obvious what has changed.

File: parsers/go.c [hunk 1]
```diff
@@ -151,6 +159,17 @@ static tokenInfo *newToken (void)
 	return token;
 }
 
+static tokenInfo *copyToken (tokenInfo *other)
```
Comment (b4n) [inline parsers/go.c]: const arg maybe

**Comment by b4n** [comment]
Assuming the parsing results are the expected ones, LGTM

**Comment by masatake** [comment]
fizz and noise are passed.

**Comment by masatake** [comment]
Thanks.

**Comment by techee** [comment]
Great, thanks!

**Pull request merged**